src/BARON.jl

```diff
@@ -156,13 +156,24 @@ wrap_with_parens(x::String) = string("(", x, ")")
 # to_str(x) = (@show(x); string(float(x)))
 to_str(x) = string(x)
 
+struct UnrecognizedExpressionException <: Exception
+    msg::String
+end
+UnrecognizedExpressionException(expr::Expr) = UnrecognizedExpressionException("unrecognized expression: $expr")
+function Base.showerror(io::IO, err::UnrecognizedExpressionException)
+    print(io, "UnrecognizedExpressionException: ")
+    print(io, err.msg)
+end
+
 function to_str(c::Expr)
     if c.head == :comparison
         if length(c.args) == 3
             return join([to_str(c.args[1]), c.args[2], c.args[3]], " ")
         elseif length(c.args) == 5
             return join([c.args[1], c.args[2], to_str(c.args[3]),
                          c.args[4], c.args[5]], " ")
+        else
+            throw(UnrecognizedExpressionException(c))
```

Comment:
Could you change the message to something like ``"unrecognized comparison expression: $c``?